packages/discord.js/src/structures/ClientApplication.js

```diff
@@ -118,16 +154,24 @@ class ClientApplication extends Application {
     } else {
       this.roleConnectionsVerificationURL ??= null;
     }
+  }
 
-    /**
-     * The owner of this OAuth application
-     * @type {?(User|Team)}
-     */
-    this.owner = data.team
-      ? new Team(this.client, data.team)
-      : data.owner
-      ? this.client.users._add(data.owner)
-      : this.owner ?? null;
+  /**
+   * The owner of this application.
+   * @type {?(User|Team)}
+   * @readonly
+   */
+  get owner() {
```

Comment:
This is technically a breaking change.
I'm also not sure why this is being changed, the behavior looks the same (except it now being read-only).

Reply:
I could revert this whole change, actually. It was just something lingering from when there was a possible overwrite due to using two different routes that contained potentially different data.